src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/SourceGenerators/IncrementalValueProviderExtensions.cs

```diff
@@ -13,13 +13,13 @@ namespace Microsoft.NET.Sdk.Razor.SourceGenerators
 {
     internal static class IncrementalValuesProviderExtensions
     {
-        internal static IncrementalValueProvider<T> WithLambdaComparer<T>(this IncrementalValueProvider<T> source, Func<T, T, bool> equal)
+        internal static IncrementalValueProvider<T> WithLambdaComparer<T>(this IncrementalValueProvider<T> source, Func<T?, T?, bool> equal)
```

Comment:
Why do we need these changes?

Reply:
IEqualityComparer<T>.Equals has signature with `T?` (which became needed in net80, before the SourceGenerators were built just for netstandard20), so the delegates that implement that in LambdaComparer also need be prepared to get nulls